### DISCUSSION TITLE
HEIMAN HS3SA detected as HEIMAN HS1SA

### DISCUSSION THREAD

**Discussion by etatto** [Koenkk/zigbee2mqtt#6790] [Devices]
Hi!

I have added recently to my zigbee2mqtt network 5 smoke detectors from HEIMAN: on the box and on the label inside the battery compartment, it is clearly written HS3SA-E, and visually they are HS3SA. But at pairing, they are detected as HEIMAN HS1SA, I think because they send "SmokeSensor-EM" as modelId, which is linked to HS1SA in devices.js and not to HS3SA (or perhaps I misunderstood how the detection is done).

So my question: is it confirm by "real life" case :wink: that a HS1SA can send "SmokeSensor-EM" for its modelId? If it is not the case, maybe "SmokeSensor-EM" should be moved to HS3SA instead in device.js.

(On a side note, the picture found on the supported devices list and the frontend is not correct I think for HS3SA: this smoke detector and its mounting piece are fully cylindrical, the HS1SA is more square with rounded egdes and its mounting piece is like a hat :smile:. A correct picture can be found here: https://zigbee.blakadder.com/Heiman_HS3SA.html)

BR.

**Comment by Koenkk** [comment]
Added in https://github.com/Koenkk/zigbee-herdsman-converters/pull/272 by @arteck , can you confirm it's a HS1SA or HS3SA?

**Reply by etatto** [reply]
No reply so far from @arteck, what is your point of view @Koenkk concerning HS3SA "SmokeSensor-EM" detected as HS1SA?
I think, as I'm curently migrating/re-pairing all zigbee devices  to my new Slaesh's Stick, I will change localy heiman.js to do the proper detection...

**Reply by Koenkk** [reply]
Marked it as a HS3SA now. 
Changes will be available in the dev branch in a few hours from now. (https://www.zigbee2mqtt.io/how_tos/how-to-switch-to-dev-branch.html)

**Reply by etatto** [reply]
I am already (and always ;)) on the dev branch, I will make a pull as soon I see your update available :-D.
Thanks.
